script.recap/resources/lib/core.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+import xbmc
+import xbmcaddon
+import xbmcgui
+
+# Import the common settings
+from settings import log
+from settings import Settings
+from recap import Recap
+from viewer import Viewer
+
+ADDON = xbmcaddon.Addon(id='script.recap')
+
+
+#########################
+# Main
+#########################
+def runRecap(previous=False):
+    log("runRecap: Started")
+
+    # First check to see if we have a TV Show already selected
+    showName = None
+    if xbmc.getInfoLabel("ListItem.dbtype") in ['tvshow', 'season', 'episode']:
```

Comment:
Also a general note (feel free to ignore): You are doing these "if x in y" checks quite often. Those checks perform a lot better when "y" is a set, and not a list. ( ---> if x in {a, b, c} )